### ISSUE TITLE
DynamoDB listener, forward_request, the JSON object must be str, not 'bytes

### ISSUE THREAD

**Issue by vaclav-krajicek** [localstack/localstack#463]
Hi,
there is an issue in dynamodb_listener.py", line 29, in forward_request related to JSON message encoding. It is exhibited by following crash while starting localstack:

```
user@machine:/tmp/localstack_install_dir$ make -C /tmp/localstack_install_dir infra
make: Entering directory '/tmp/localstack_install_dir'
(. .venv/bin/activate; exec bin/localstack start)
Starting local dev environment. CTRL-C to quit.
Starting mock SQS (http port 4576)...
Starting mock ES service (http port 4578)...
Starting mock Firehose service (http port 4573)...
Starting mock DynamoDB Streams service (http port 4570)...
Starting mock SNS (http port 4575)...
Starting mock CloudFormation (http port 4581)...
Starting mock CloudWatch (http port 4582)...
Starting mock Redshift (http port 4577)...
Starting mock Lambda service (http port 4574)...
Starting mock DynamoDB (http port 4569)...
Starting mock SSM (http port 4583)...
Starting mock S3 (http port 4572)...
Starting mock Route53 (http port 4580)...
Starting mock API Gateway (http port 4567)...
Starting mock Kinesis (http port 4568)...
Starting mock SES (http port 4579)...
Starting local Elasticsearch (http port 4571)...
2017-11-15T09:56:13:ERROR:localstack.services.generic_proxy: Error forwarding request: the JSON object must be str, not 'bytes' Traceback (most recent call last):
  File "/tmp/localstack_install_dir/localstack/services/generic_proxy.py", line 161, in forward
    path=path, data=data, headers=forward_headers)
  File "/tmp/localstack_install_dir/localstack/services/dynamodb/dynamodb_listener.py", line 29, in forward_request
    data = json.loads(data)
  File "/usr/lib/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```

I traced back the problem to commit b8d8d8bf6763c8ec6655415793d8c8fbd57269c7.
I think that some simple conversion of data using to_str() before passing it to JSON parser in forward_request might fix the issue.

My current workaround is to use older commits of localstack than the one I mentioned above.

Thanks

**Comment by whummer** [comment]
Thanks for reporting @vaclav-krajicek 

> I think that some simple conversion of data using to_str() before passing it to JSON parser in forward_request might fix the issue.

Would you be able to create a quick pull request with that fix? Thanks!

**Comment by vaclav-krajicek** [comment]
Yes, I would.

**Comment by whummer** [comment]
Fixed in #473
